Updated Get Started with Messaging +

dawnkelly09 opened the pull request:
### Description

Please explain the changes this PR addresses here.

-> Moves the original Get Started with Messaging content (the messageSender.sol/messageReceveiver.sol demo) to Get Started with the Solidity SDK
-> Updates Get Started with Messaging with an example using core protocol to construct and publish a message
-> Adds some Next Steps to these pages to help with developer journey
-> moves any images/snippets to reflect the above structure

TODO:
- ID additional Next Steps to add to Get Started with Solidity SDK
- Add link to Fetch the Signed VAA guide (needs written)

### Checklist

- [ x] **Required** - I have added a label to this PR 🏷️
- [x ] **Required** - I have run my changes through Grammarly
- [ n/a this is all staging ] If pages have been moved, I have created redirects in the `wormhole-mkdocs` repo

eshaben commented on products/messaging/get-started.md:
```diff
@@ -1 +1,82 @@
+Before you begin, ensure you have the following:
+
+- Completed the [Get Started with the TypeScript SDK](/docs/tools/solidity-sdk/get-started){target=\_blank} guide. You should have a working project, the SDK installed, and an initialized `Wormhole` instance
+- [`ethers.js`](https://docs.ethers.org/v6/getting-started/){target=\_blank} installed (this example uses version 6)
```
We should consider putting version variables into the `variables.yml` file

dawnkelly09 replied:
Opened a ticket for this

eshaben commented on products/messaging/get-started.md:
```diff
@@ -1 +1,82 @@
+
+Before you begin, ensure you have the following:
+
+- Completed the [Get Started with the TypeScript SDK](/docs/tools/solidity-sdk/get-started){target=\_blank} guide. You should have a working project, the SDK installed, and an initialized `Wormhole` instance
```
Can't suggest changes for some reason on this line. The link is pointing to the Solidity SDK instead of TypeScript

eshaben commented on products/messaging/get-started.md:
```diff
@@ -1 +1,82 @@
+- Completed the [Get Started with the TypeScript SDK](/docs/tools/solidity-sdk/get-started){target=\_blank} guide. You should have a working project, the SDK installed, and an initialized `Wormhole` instance
+- [`ethers.js`](https://docs.ethers.org/v6/getting-started/){target=\_blank} installed (this example uses version 6)
+- [Foundry](https://book.getfoundry.sh/getting-started/installation){target=\_blank} for encrypting your private key
+- A small amount of Sepolia ETH for gas fees. You can use the [Sepolia PoW Faucet](https://sepolia-faucet.pk910.de/){target=\_blank} if you need tokens
```
```suggestion
- A small amount of [Sepolia ETH](https://sepolia-faucet.pk910.de/){target=\_blank} for gas fees
```

eshaben commented on products/messaging/get-started.md:
```diff
@@ -1 +1,82 @@
+
+## Next Steps
+
+- [Get Started with the Solidity SDK](/docs/tools/solidity-sdk/get-started/){target=\_blank} - follow this guide to create basic message sender and receiver contracts using the Wormhole Solidity SDK, deploy the contracts, and use them to send a message across blockchains
```
```suggestion
- [**Get Started with the Solidity SDK**](/docs/tools/solidity-sdk/get-started/){target=\_blank} - follow this guide to create basic message sender and receiver contracts using the Wormhole Solidity SDK, deploy the contracts, and use them to send a message across blockchains
```

eshaben commented on products/messaging/get-started.md:
```diff
@@ -1 +1,82 @@
+
+- [Get Started with the Solidity SDK](/docs/tools/solidity-sdk/get-started/){target=\_blank} - follow this guide to create basic message sender and receiver contracts using the Wormhole Solidity SDK, deploy the contracts, and use them to send a message across blockchains
+
+- [Fetch the Signed VAA](TODO WIP){target=\_blank} - whether your message is sent using core protocol or custom smart contracts, follow this guide to use your transaction information to fetch a signed VAA and decode the payload
```
```suggestion
- [**Fetch the Signed VAA**](TODO WIP){target=\_blank} - whether your message is sent using core protocol or custom smart contracts, follow this guide to use your transaction information to fetch a signed VAA and decode the payload
```

eshaben commented on tools/solidity-sdk/get-started.md:
```diff
@@ -1 +1,150 @@
+- [Testnet CELO for Celo Alfajores](https://faucet.celo.org/alfajores){target=\_blank}
+- Wallet private key
+
+## Install and Set Up Project
```
What about for people that don't want to set up a project, maybe they already have one. They're just interested in getting started with the SDK specifically. We're not including any instructions for them to install the SDK

eshaben commented on tools/solidity-sdk/get-started.md:
```diff
@@ -1 +1,150 @@
+
+## Next Steps
+
+- [Fetch the Signed VAA](TODO WIP){target=\_blank} - whether your message is sent using core protocol or custom smart contracts, follow this guide to use your transaction information to fetch a signed VAA and decode the payload
```
```suggestion
- [**Fetch the Signed VAA**](TODO WIP){target=\_blank} - whether your message is sent using core protocol or custom smart contracts, follow this guide to use your transaction information to fetch a signed VAA and decode the payload
```

eshaben reviewed (commented):
Nice work!

Although I know the private key setup is important, I wish it didn't take up so much real estate on the page.  The two sections on encryption is just as long as the actual meat and potatoes of the guide  🤷‍♀️ 

For the Solidity SDK get started page, I'd like it to refocus is a little bit on the actual SDK rather than the example. I think obviously the example is important to showcase what you can do with the SDK, but the introduction doesn't mention the SDK, there is no installation instructions for the SDK

dawnkelly09 commented:
Solidity Get Started: use smart contracts to send testnet USDC from AVAX Fuji to Celo testnet
- install Solidity SDK
- create token sender & recvr contracts
- deployment script & instructions
- transfer script & instructions

Messaging Get Started: use TypeScript SDK & core protocol to publish a message to Sepolia network
- install TS SDK
- signMessage script and instructions
- main function to send message
- returns VAA identifiers for use in future "Fetch Signed VAA" guide

eshaben commented on tools/solidity-sdk/get-started.md:
So the Solidity SDK is definitely a bit different from the TypeScript SDK, haha obviously, but we still need to treat it as tool documentation. So if I put myself in the shoes of someone new to the SDK and trying to get started with it, I'm going to want to know how to install the SDK, how to import the different contracts, what contracts are available to me to import (high-level), and then an example

eshaben replied:
Ok, so iof you put yourself in the shoes of someone who wants to build a contract-based solution and you just came across the Solidity SDK with no real context about what all it provides, what would you expect to find on the get started page?

For me, I think it'd be nice to see what's available on a high-level. If not on this page, the other option is the reference page since the tools don't have an overview page. Regardless, I think we should most definitely add a high-level list or link to one because this would be the type of question I would hope to find an answer to

eshaben replied:
Maybe in the introduction we need to elaborate a little bit more in this sentence: "providing all necessary Solidity interfaces along with useful libraries and tools for testing."

That's pretty vague and yes, it's an introduction, but what does this mean?

dawnkelly09 commented on products/messaging/get-started.md:
````diff
@@ -1,149 +1,121 @@
+        }
+        ```
+    
+    4. Use the following command to install the Wormhole TypeScript SDK:
````
I went back and forth on this so happy to have a second opinion on it.

dawnkelly09 commented on products/messaging/get-started.md:
````diff
@@ -1,149 +1,121 @@
+        ```
 
-The `chains.json` configuration defines properties for supported chains, including the Wormhole relayer addresses, RPC URLs, and chain IDs, and provides this information to the deployment scripts when you run them.
+## Prepare to Sign Messages with Your Encrypted Key
````
I'll go back and read that and make some tweaks. Thanks!

ilariae commented on products/messaging/get-started.md:
```diff
@@ -1,149 +1,121 @@
-## Introduction
-
-Wormhole's messaging protocol simplifies sending data, triggering events, and initiating transactions across blockchain networks. This guide demonstrates how to configure and deploy contracts to send messages from Avalanche Fuji to Celo Alfajores.
+Wormhole's core functionality allows you to send any data packet from one supported chain to another. This guide will demonstrate publishing your first simple, arbitrary data message from an EVM environment source chain using the Wormhole TypeScript SDK's core messaging capabilities. 
```
```suggestion
Wormhole's core protocol allows you to send any data packet from one supported chain to another. This guide will demonstrate publishing your first simple, arbitrary data message from an EVM environment source chain using the Wormhole TypeScript SDK's core messaging capabilities. 
```

ilariae commented on products/messaging/get-started.md:
````diff
@@ -1,149 +1,121 @@
     ```
 
-2. Enter your Foundry keystore password in the terminal when prompted
+     The `signMessage.ts` script creates an `ethers.js` provider, decrypts the Foundry keystore, connects to a wallet, and creates an full signer for the EVM environment. It returns the `signer` and `provider` for use in your messaging script.
````
i actually like having file names in backticks, it makes it easy to spot what file we're working on

eshaben replied:
It's referring to the library, not a file name

eshaben replied:
But yes, file names should be in backticks

ilariae reviewed (commented):
left some minor comments and I agree with most of erin's comments, the rest looks good to me

dawnkelly09 commented:
I believe this is ready for review again. Primary changes:
- simplified both projects through refactoring (less files, etc)
- updated handling of keys to be like Token Bridge example
- more generic when mentioning chains in prereqs per feedback

I think this closes the gaps from the previous version. Look forward to feedback

eshaben commented on tools/solidity-sdk/get-started.md:
```diff
@@ -1 +1,145 @@
+
+The [Wormhole Solidity SDK](https://github.com/wormhole-foundation/wormhole-solidity-sdk){target=\_blank} provides Solidity interfaces, prebuilt contracts, and testing tools to help Solidity developers build on-chain Wormhole integrations via smart contracts. You can use the [Wormhole TypeScript SDK](/docs/tools/typescript-sdk/get-started/){target=\_blank} for off-chain integrations without writi ...
+
+## Install the Wormhole Solidity SDK
```
To align with the TypeScript SDK page

eshaben commented on .snippets/code/tools/solidity-sdk/get-started/solidity-sdk-1.sol:
```diff
@@ -0,0 +1,61 @@
+import "lib/wormhole-solidity-sdk/src/WormholeRelayerSDK.sol";
+import "lib/wormhole-solidity-sdk/src/interfaces/IERC20.sol";
+
+// Assign the contract to the TokenSender role inherited from TokenBase
```
I'm not sure assign the role is the correct way to phrase this. I'd update these comments to say "Extend" or "Inherit"

For example: `// Extend the TokenSender contract inherited from TokenBase`

Ideally this change be made in the steps in the guide and also the comments here and in solidity-sdk-2.sol

eshaben commented on .snippets/code/tools/solidity-sdk/get-started/solidity-sdk-1.sol:
```diff
@@ -0,0 +1,61 @@
+// Assign the contract to the TokenSender role inherited from TokenBase
+contract CrossChainSender is TokenSender {
+    uint256 constant GAS_LIMIT = 250_000;
+    // Initialize the contract with the Wormhole relayer, token bridge, and wormhole address
```
```suggestion
    // Initialize the contract with the Wormhole relayer, Token Bridge,
    // and Wormhole Core Contract addresses
```

eshaben commented on .snippets/code/tools/solidity-sdk/get-started/solidity-sdk-1.sol:
```diff
@@ -0,0 +1,61 @@
+        address _wormhole
+    ) TokenBase(_wormholeRelayer, _tokenBridge, _wormhole) {}
+
+    // Calculate the estimated cost for multichain token transfer
+    // using the wormholeRelayer to get the delivery cost and add the message fee
```
```suggestion
    // Calculate the estimated cost for multichain token transfer using
    // the wormholeRelayer to get the delivery cost and add the message fee
```

If you look at the rendered snippet, it's kind of weird because certain things get pushed to the next line but don't have the `//` on that forced line break. So this just fixes that

eshaben commented on .snippets/code/tools/solidity-sdk/get-started/solidity-sdk-2.sol:
```diff
@@ -0,0 +1,42 @@
+import "lib/wormhole-solidity-sdk/src/WormholeRelayerSDK.sol";
+import "lib/wormhole-solidity-sdk/src/interfaces/IERC20.sol";
+
+// Assign the contract to the TokenReceiver role inherited from TokenBase
```
Just flagging the "Assign"/"role" thing

eshaben commented on .snippets/code/tools/solidity-sdk/get-started/solidity-sdk-2.sol:
```diff
@@ -0,0 +1,42 @@
+
+// Assign the contract to the TokenReceiver role inherited from TokenBase
+contract CrossChainReceiver is TokenReceiver {
+    // Initialize the contract with the Wormhole relayer, token bridge, and wormhole address
```
```suggestion
    // Initialize the contract with the Wormhole relayer, Token Bridge,
    // and Wormhole Core Contract addresses
```

eshaben commented on .snippets/code/tools/solidity-sdk/get-started/solidity-sdk-3.ts:
```diff
@@ -0,0 +1,154 @@
+import path from 'path';
+import readlineSync from 'readline-sync';
+import { fileURLToPath } from 'url';
+import {
+  wormhole,
+  amount,
+  Wormhole,
+  ChainContext,
+  Signer,
+  Chain,
+  Network,
+  ChainAddress,
+  isTokenId,
+  TokenId,
+  chainToChainId,
+} from '@wormhole-foundation/sdk';
```
```suggestion
import { wormhole, chainToChainId } from '@wormhole-foundation/sdk';
```

eshaben commented on .snippets/code/tools/solidity-sdk/get-started/solidity-sdk-3.ts:
```diff
@@ -0,0 +1,154 @@
+    const sendChain = wh.getChain(AVALANCHE_CHAIN_NAME);
+    const rcvChain = wh.getChain(CELO_CHAIN_NAME);
+
+    // Create an ethers.Signer wallet with your EVM private key
+    // using locally scoped credentials. EVM_PRIVATE_KEY must
+    // be loaded securely beforehand, for example via a keystore,
+    // secrets manager, or environment variables
+    // (not recommended)
```
```suggestion
    // The EVM_PRIVATE_KEY value must be loaded securely beforehand,
    // for example via a keystore, secrets manager, or environment variables
    // (not recommended)
```

eshaben commented on tools/solidity-sdk/get-started.md:
````diff
@@ -1 +1,146 @@
+```bash
+forge install wormhole-foundation/wormhole-solidity-sdk
+```
+
+Next, review the following sections for more information on SDK components and features, or jump to [How It Works](#how-it-works) for a build-along demo project.
+
+## Key Components and Features
+
````
```suggestion

## Key Components

```

Then the rest of this section can be like the example I shared in the main comment on the PR. Very brief, cut out any extra stuff. Just quick like hey here's what's available type deal

eshaben commented on tools/solidity-sdk/get-started.md:
```diff
@@ -1 +1,146 @@
+
+The following key components and features work together to make your on-chain Wormhole integration easier to build.
+
+### Wormhole Relayer Integration
```
Delete this section

eshaben commented on tools/solidity-sdk/get-started.md:
```diff
@@ -1 +1,146 @@
+        - [**`TokenBase.sol`**](https://github.com/wormhole-foundation/wormhole-solidity-sdk/blob/b9e129e65d34827d92fceeed8c87d3ecdfc801d0/src/TokenBase.sol){target=\_blank}: Uses `TokenReceiver` and `TokenSender` contracts to define functions for transferring tokens.
+        - [**`CCTPBase.sol`**](https://github.com/wormhole-foundation/wormhole-solidity-sdk/blob/b9e129e65d34827d92fceeed8c87d3ecdfc801d0/src/CCTPBase.sol){target=\_blank}: Uses `CCTPSender` and `CCTPReceiver` contracts to define functions for transferring USDC.
+
+### Solidity Interfaces
```
Delete

eshaben commented on tools/solidity-sdk/get-started.md:
```diff
@@ -1 +1,146 @@
+    - [**`IWormholeReceiver.sol`**](https://github.com/wormhole-foundation/wormhole-solidity-sdk/blob/b9e129e65d34827d92fceeed8c87d3ecdfc801d0/src/interfaces/IWormholeReceiver.sol){target=\_blank}: Defines the `receiveWormholeMessages` function.
+    - [**`IWormholeRelayer.sol`**](https://github.com/wormhole-foundation/wormhole-solidity-sdk/blob/b9e129e65d34827d92fceeed8c87d3ecdfc801d0/src/interfaces/IWormholeRelayer.sol){target=\_blank}: Defines key structs and functions to identify, send, and deliver messages and follow the progress of transactions.
+
+### Helpful Constants
```
Delete

eshaben commented on tools/solidity-sdk/get-started.md:
```diff
@@ -1 +1,146 @@
+    - [**Circle CCTP Domain IDs**](https://github.com/wormhole-foundation/wormhole-solidity-sdk/blob/b9e129e65d34827d92fceeed8c87d3ecdfc801d0/src/CCTPAndTokenBase.sol){target=\_blank}: Generated list of defined CCTP domain ID's to ensure USDC transfers use the correct domain for a given chain. 
+    - [**`chainConsts.ts`**](https://github.com/wormhole-foundation/wormhole-solidity-sdk/blob/75ddcec06ffe9d62603d023357caa576c5ea101c/gen/chainConsts.ts){target=\_blank}: Returns values to identify properties and contract addresses for each supported chain.
+
+### Testing Utilities
```
Delete

eshaben commented on tools/solidity-sdk/get-started.md:
```diff
@@ -1 +1,146 @@
+
+The Wormhole Solidity SDK also includes a robust set of [testing utilities](https://github.com/wormhole-foundation/wormhole-solidity-sdk/tree/75ddcec06ffe9d62603d023357caa576c5ea101c/test) for simulating message and token transfers to test your build.
+
+## Prerequisites
+
+The following demo illustrates the use of Wormhole Solidity SDK-based smart contracts to send testnet USDC between supported chains.
+
```
```suggestion
## Example Usage

The following demo illustrates the use of Wormhole Solidity SDK-based smart contracts to send testnet USDC between supported chains.

### Prerequisites
```

haha change my mind on this, sorry

eshaben commented on tools/solidity-sdk/get-started.md:
```diff
@@ -1 +1,146 @@
+- [Node.js and npm](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm){target=\_blank} installed
+- [TypeScript](https://www.typescriptlang.org/download/){target=\_blank} installed
+- [Foundry](https://book.getfoundry.sh/getting-started/installation){target=\_blank} installed
+- Testnet tokens for two supported chains. This example uses [testnet AVAX for Avalanche Fuji](https://core.app/tools/testnet-faucet/?subnet=c&token=c){target=\_blank} and [testnet CELO for Celo Alfajores](https://faucet.celo.org/alfajores){target=\_blank} and can be adapted to any supported chains.
```
```suggestion
- Testnet tokens for two supported chains. This example uses [testnet AVAX for Avalanche Fuji](https://core.app/tools/testnet-faucet/?subnet=c&token=c){target=\_blank} and [testnet CELO for Celo Alfajores](https://faucet.celo.org/alfajores){target=\_blank} and can be adapted to any supported chains
```

eshaben commented on tools/solidity-sdk/get-started.md:
```diff
@@ -1 +1,146 @@
+- [Foundry](https://book.getfoundry.sh/getting-started/installation){target=\_blank} installed
+- Testnet tokens for two supported chains. This example uses [testnet AVAX for Avalanche Fuji](https://core.app/tools/testnet-faucet/?subnet=c&token=c){target=\_blank} and [testnet CELO for Celo Alfajores](https://faucet.celo.org/alfajores){target=\_blank} and can be adapted to any supported chains.
+- [USDC testnet tokens](https://faucet.circle.com/){target=\_blank} on your source chain for cross-chain transfer
+
```
```suggestion

### Set Up a Project

```

eshaben commented on tools/solidity-sdk/get-started.md:
````diff
@@ -1 +1,146 @@
+    npm init -y && npm install @wormhole-foundation/sdk ethers -D tsx typescript
+    ```
+
+## Create and Deploy Contracts
````
```suggestion
### Create and Deploy Contracts
```

eshaben commented on tools/solidity-sdk/get-started.md:
````diff
@@ -1 +1,146 @@
+    --8<-- "code/tools/solidity-sdk/get-started/solidity-sdk-1.sol"
+    ```
+
+    This contract assigns `CrossChainSender` to the `TokenSender` role, initializes the contract, calculates a cost of transfer estimate, defines transfer parameters, and initiates the transfer using the `sendTokenWithPayloadToEvm` function from `WormholeRelayer`.
````
```suggestion
    This contract extends `TokenSender`, gaining access to its functionality. It initializes the contract with the required addresses, calculates estimated transfer costs, defines transfer parameters, and initiates the transfer using the `sendTokenWithPayloadToEvm` function from `WormholeRelayer`.
```

Instead of "assign" and "role"

eshaben commented on tools/solidity-sdk/get-started.md:
````diff
@@ -1 +1,146 @@
+    --8<-- "code/tools/solidity-sdk/get-started/solidity-sdk-2.sol"
+    ```
+
+    This contract assigns `CrossChainReceiver` to the `TokenReceiver` role, initializes the contract, receives the payload and tokens, verifies the transfer is from a registered sender, decodes the recipient address, and transfers the tokens to the recipient.
````
```suggestion
    This contract extends `TokenReceiver`, gaining access to its functionality. It initializes the contract with the required addresses, receives the payload and tokens, verifies the transfer is from a registered sender, decodes the recipient address, and transfers the tokens to the recipient.
```

Same thing here

eshaben commented on tools/solidity-sdk/get-started.md:
```diff
@@ -1 +1,146 @@
+
+3. Deploy the contracts using your preferred deployment method. Make sure you deploy `CrossChainSender.sol` to your desired source chain and `CrossChainReceiver.sol` to the target chain. Save the deployed contract addresses for each contract. You will need them for your transfer script.
+
+## Create Your Transfer Script
```
```suggestion
##  Use Contracts to Transfer USDC
```

What are we doing with the script though? What's the important part here

eshaben commented on tools/solidity-sdk/get-started.md:
````diff
@@ -1 +1,146 @@
+    --8<-- "code/tools/solidity-sdk/get-started/solidity-sdk-3.ts"
+    ```
+
+    This script defines sender and receiver contract addresses, fetches needed ABI information, creates a connected signer, converts decimals, qoutes transfer cost, and initiates the token transfer.
````
```suggestion
    This script defines the sender and receiver contract addresses, fetches the necessary ABI information, creates a connected signer, converts decimals, calculates the estimated transfer cost, and initiates the token transfer.
```

eshaben commented on products/messaging/get-started.md:
````diff
@@ -1,149 +1,101 @@
     ```
 
-2. Enter the password you wish to use to decrypt your private key at the prompt. You will not see the password in the terminal as you type:
+4. Install the TypeScript SDK:
````
```suggestion
4. Install the [TypeScript SDK](/docs/tools/typescript-sdk/get-started/){target=\_blank}:
```

eshaben reviewed (commented):
So, I went to update the Solidity SDK page so the key components section was something like this:

<details>
<summary> Updated key components section </summary>
## Key Components

The following key components and features work together to make your on-chain Wormhole integration easier to build.

??? interface "Base contracts"

    Leverage base contracts to send messages, tokens, and USDC.

    - [**`Base.sol`**](https://github.com/wormhole-foundation/wormhole-solidity-sdk/blob/main/src/Base.sol){target=\_blank}: Uses Wormhole interfaces to authorize and verify a registered sender.
    - [**`TokenBase.sol`**](https://github.com/wormhole-foundation/wormhole-solidity-sdk/blob/main/src/TokenBase.sol){target=\_blank}: Uses `TokenReceiver` and `TokenSender` contracts to define functions for transferring tokens.
    - [**`CCTPBase.sol`**](https://github.com/wormhole-foundation/wormhole-solidity-sdk/blob/main/src/CCTPBase.sol){target=\_blank}: Uses `CCTPSender` and `CCTPReceiver` contracts to define functions for transferring USDC.

??? interface "Interfaces"

    Use interfaces to ensure consistent interactions with the protocol regardless of the supported chain you use.

    - [**`ITokenBridge.sol`**](https://github.com/wormhole-foundation/wormhole-solidity-sdk/blob/main/src/interfaces/ITokenBridge.sol){target=\_blank}: Defines key structs and functions for token attestation, wrapping and transferring tokens, monitoring transaction progress.
    - [**CCTP Interfaces**](https://github.com/wormhole-foundation/wormhole-solidity-sdk/tree/main/src/interfaces/CCTPInterfaces){target=\_blank}: A set of interfaces for USDC transfers via CCTP for sending, relaying, and receiving messages and tokens.
    - [**`IWormholeReceiver.sol`**](https://github.com/wormhole-foundation/wormhole-solidity-sdk/blob/main/src/interfaces/IWormholeReceiver.sol){target=\_blank}: Defines the `receiveWormholeMessages` function.
    - [**`IWormholeRelayer.sol`**](https://github.com/wormhole-foundation/wormhole-solidity-sdk/blob/main/src/interfaces/IWormholeRelayer.sol){target=\_blank}: Defines key structs and functions to identify, send, and deliver messages and follow the progress of transactions.

??? interface "Constants"

    Auto-generated Solidity constants help avoid manual entry errors and ensure consistent delivery.

    - [**Wormhole Chain IDs**](https://github.com/wormhole-foundation/wormhole-solidity-sdk/blob/main/src/Chains.sol){target=\_blank}: Generated list of Wormhole Chain IDs for supported chains.
    - [**Circle CCTP Domain IDs**](https://github.com/wormhole-foundation/wormhole-solidity-sdk/blob/main/src/CCTPAndTokenBase.sol){target=\_blank}: Generated list of defined CCTP domain IDs to ensure USDC transfers use the correct domain for a given chain. 
    - [**`chainConsts.ts`**](https://github.com/wormhole-foundation/wormhole-solidity-sdk/blob/main/gen/chainConsts.ts){target=\_blank}: Returns values to identify properties and contract addresses for each supported chain.
</details>

But, while I was doing that I noticed you were anchoring the links to certain commits. Ideally they would have a release process and we could anchor it to a release tag, but that's not the case. So I tried to anchor it to main and then I realized they've done a lot of work on the SDK and things have changed quite a bit. Soooooo... we'll need to make some updates to align with the latest changes. @ilariae do you know if more changes are coming?

For now,  maybe we can ship with the anchored commits. But ideally they'd at least be all using the same commit

ilariae commented:
> So, I went to update the Solidity SDK page so the key components section was something like this:
> 
> Updated key components section
> But, while I was doing that I noticed you were anchoring the links to certain commits. Ideally they would have a release process and we could anchor it to a release tag, but that's not the case. So I tried to anchor it to main and then I realized they've done a lot of work on the SDK and things have changed quite a bit. Soooooo... we'll need to make some updates to align with the latest changes. @ilariae do you know if more changes are coming?
> 
> For now, maybe we can ship with the anchored commits. But ideally they'd at least be all using the same commit

I dont know what changes are coming but the solidity SDK improvements have been audited 2 days ago from what i know so should be coming in june sometime

eshaben commented on products/messaging/get-started.md:
```diff
@@ -1,149 +1,101 @@
 ## Next Steps
 
-<!--TODO: links to other guides and tutorials-->
+- [**Get Started with Token Bridge**](/docs/products/token-bridge/get-started/){target=\_blank}: follow this guide to start working with multichain token transfers using Wormhole Token Bridge's lock and mint mechanism to send tokens across chains.
 
+- [**Get Started with the Solidity SDK**](/docs/tools/solidity-sdk/get-started/){target=\_blank}: smart contract developers can follow this on-chain integration guide to use Wormhole Solidity SDK-based sender and receiver contracts to send testnet USDC across chains.
```
```suggestion
- [**Get Started with Token Bridge**](/docs/products/token-bridge/get-started/){target=\_blank}: Follow this guide to start working with multichain token transfers using Wormhole Token Bridge's lock and mint mechanism to send tokens across chains.

- [**Get Started with the Solidity SDK**](/docs/tools/solidity-sdk/get-started/){target=\_blank}: Smart contract developers can follow this on-chain integration guide to use Wormhole Solidity SDK-based sender and receiver contracts to send testnet USDC across chains.
```
